backends-velox/src/main/scala/org/apache/gluten/backendsapi/velox/VeloxSparkPlanExecApi.scala

```diff
@@ -914,4 +915,25 @@ class VeloxSparkPlanExecApi extends SparkPlanExecApi {
       child: Seq[SparkPlan]): ColumnarRangeBaseExec =
     ColumnarRangeExec(start, end, step, numSlices, numElements, outputAttributes, child)
 
+  override def genShuffleRead(plan: SparkPlan): SparkPlan = {
+    if (!VeloxConfig.get.veloxResizeBatchesShuffleInput) return plan
```

Comment:
can we use another config for this?
Besides, have you checked TPCDS performance on this PR? It seems suitable only for specific scenarios.

Reply:
seems there are performance downgrade. I'll investigate later